NYPL/authoritydata_updater
# Support graphic material DR-531

Hey @emu47 

This branch is used to parse-and-post records for the [Thesaurus of Graphic Material](http://id.loc.gov/vocabulary/graphicMaterials.html). As you know, this is know, this data set is not available in JSON format, but we can parse the n-tripple (.nt) file to generate the hashes we post to Solr.  The upside is **the new class `TrippleToSolrDoc`, can be used in the future to parse other vocabularies only available in .nt files.**

## Testing Locally

1.  Download and unzip the [the graphic materials nt file](http://id.loc.gov/static/data/downloads/vocabularygraphicMaterials.nt.both.zip) and put it in the root of this project. It will get mounted into the container.
2.  `docker-compose build updater --no-cache`. (There are new gems)
3.  `docker compose run updated bash` (This brings up Solr, and drops you off in a shell session).
4.  Confirm there are no documents in your solr:

-   <http://localhost:8983/solr/admin/cores?action=STATUS&core=authoritydata>
-   <http://localhost:8983/solr/admin/>

5.  (Inside the container) go to `/opt/authoritydata_udpater/` and run:


>ruby /opt/authoritydata_udpater/authoritydata_updater.rb --vocabulary graphic_materials --source /opt/authoritydata_udpater/graphicMaterials.both.nt --solrUrl $SOLR_URL --username $USERNAME --password $PASSWORD


6.  Make some coffee - this parses a 493,314 line file.
7.  By the end of your coffee break it will post to solr.
8.  Check those Solr endpoints again, you'll see the documents in there.

## Once you approve this, let's "deploy" it together.

I'd like your help deleting-by-query the existing Thesaurus of Graphic Material,
then we can post this together. We'll obviously start on the QA Solr.
